modules/migration-about-configuring-proxies.adoc

```diff
@@ -0,0 +1,124 @@
+
+=== TCP proxy setup for DVM
+
+You can set up direct connection between the source and the target cluster through a TCP proxy and configure the `stunnel_tcp_proxy` variable in the `MigrationController` CR use the proxy:
+
+[source, yaml]
+----
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigrationController
+metadata:
+  name: migration-controller
+  namespace: openshift-migration
+spec:
+  [...]
+  stunnel_tcp_proxy: http://username:password@ip:port
+----
+
+Direct volume migration (DVM) only supports Basic Authentication for the proxy. Moreover, DVM only works behind proxies which can tunnel a TCP connection transparently. HTTP/HTTPS proxies in man-in-the-middle mode will not work. The existing cluster-wide proxies may or may not support this behavior. As a result, the proxy settings for DVM are intentionally kept different from usual proxy configuration in {mtc-short}.
+
+=== Why use a TCP proxy instead of an HTTP/HTTPS proxy?
+
+DVM is enabled by running Rsync between the source and the target cluster over an OpenShift Route.  Traffic is encrypted using Stunnel, which in itself is a TCP proxy. The Stunnel running on the source cluster initiates a TLS connection with the target Stunnel and transfers data over an encrypted channel.
+
+Cluster-wide HTTP/HTTPS proxies in OpenShift are usually configured in man-in-the-middle mode where they negotiate their own TLS session with the outside servers. However, this does not work with Stunnel. Stunnel requires that its TLS session be untouched by the proxy, essentially making the proxy a transparent tunnel which simply forwards the TCP connection as-is. Therefore, a TCP proxy must be used.
+
+=== Known Issue
+
+.Migration fails with error `Upgrade request required`
+
+The migration Controller uses the SPDY protocol to execute commands within remote Pods. If the remote cluster is behind a proxy or a firewall that does not support the SPDY protocol, the migration controller  fails to execute remote commands. The migration fails with the error message `Upgrade request required`.
+Workaround: Use a proxy that supports the SPDY protocol.
+
+In addition to supporting the SPDY protocol, the proxy or firewall also must pass the `Upgrade` HTTP header to the API server. The client uses this header to open a websocket connection with the API server. If the `Upgrade` header is blocked by the proxy or firewall, the migration fail with the error message `Upgrade request required`.
+Workaround: Ensure that the proxy forwards the `Upgrade` header.
+
+== Tuning Network Policies for migrations
+
+OpenShift supports restricting traffic to or from Pods using _NetworkPolicy_ or and _EgressFirewalls_ based on the network plugin used by the cluster. If any of the source namespaces involved in a migration use such mechanisms to restrict network traffic to Pods, the restrictions might inadvertently stop traffic to Rsync Pods during migration.
```

Comment:
using _NetworkPolicy_ or **and** _EgressFirewalls_ based >> is the and required?